src/pybind/mgr/orchestrator/module.py

```diff
@@ -1115,6 +1114,8 @@ def _apply_rgw(self,
             preview_only=dry_run
         )
 
+        spec.validate()  # force any validation exceptions to be caught correctly
```

Comment:
This is minor but do you mean "early" or "immediately" rather than "correctly" here?

Reply:
in this case is immediatly since the exception will be raised right away